python/delta_sharing/tests/test_reader.py

```diff
@@ -448,3 +469,186 @@ def list_table_changes(
+                    '{"fields":['
+                    '{"metadata":{},"name":"a","nullable":true,"type":"long"},'
+                    '{"metadata":{},"name":"b","nullable":true,"type":"string"}'
+                    '],"type":"struct"}'
+                )
+            )
+            add_files = [
+                AddFile(
+                    url=str(tmp_path / "pdf1.parquet"),
+                    id="pdf1",
+                    partition_values={},
+                    size=0,
+                    stats="",
+                ),
+                AddFile(
+                    url=str(tmp_path / "pdf2.parquet"),
+                    id="pdf2",
+                    partition_values={},
+                    size=0,
+                    stats="",
+                ),
+            ]
+            return ListFilesInTableResponse(
+                delta_table_version=1,
+                protocol=None,
+                metadata=metadata,
+                add_files=add_files,
+            )
+
+    reader = DeltaSharingReader(Table("table_name", "share_name", "schema_name"), RestClientMock())
+    ds = reader.to_pyarrow_dataset()
+    expected = pa_ds.dataset(paths, format="parquet")
+
+    assert expected.count_rows() == ds.count_rows()
+    assert expected.schema.equals(ds.schema, check_metadata=True)
+
+    ds_paths = [f.path for f in ds.get_fragments()]
+
+    assert len(ds_paths) == len(paths) == 2
+    assert sorted(ds_paths) == sorted(paths)
+
+
+def test_to_pyarrow_dataset_empty(tmp_path):
+    expected_schema = pa.schema([("a", pa.int64()), ("b", pa.string())])
+
+    class RestClientMock:
+        def list_files_in_table(
+            self,
+            table: Table,
+            *,
+            predicateHints: Optional[Sequence[str]] = None,
+            limitHint: Optional[int] = None,
+            version: Optional[int] = None,
+            timestamp: Optional[int] = None,
+        ) -> ListFilesInTableResponse:
+            assert table == Table("table_name", "share_name", "schema_name")
+
+            metadata = Metadata(
+                schema_string=(
+                    '{"fields":['
+                    '{"metadata":{},"name":"a","nullable":true,"type":"long"},'
+                    '{"metadata":{},"name":"b","nullable":true,"type":"string"}'
+                    '],"type":"struct"}'
+                )
+            )
+            add_files: Sequence[AddFile] = []
+
+            return ListFilesInTableResponse(
+                delta_table_version=1,
+                protocol=None,
+                metadata=metadata,
+                add_files=add_files,
+            )
+
+    reader = DeltaSharingReader(Table("table_name", "share_name", "schema_name"), RestClientMock())
+    ds = reader.to_pyarrow_dataset()
+
+    assert expected_schema.equals(ds.schema, check_metadata=True)
+    assert len([f.path for f in ds.get_fragments()]) == 0
+    assert ds.count_rows() == 0
+
+
+def test_to_pyarrow_table_non_partitioned(tmp_path):
```

Comment:
What's the difference between this test and `test_to_pyarrow_dataset`?

Reply:
thats for pyarrow dataset (lazy, faster) and this is for pyarrow table (eager)